src/acl/agent.ts

```diff
@@ -386,6 +386,21 @@ function removeAgentFromResourceRule(
     acl.accessTo,
     resourceIri
   );
+  const agentHasDefaultAccess =
+    getIriAll(rule, acl.agent).filter(
+      (originalAgent) => originalAgent === agent
+    ).length > 0;
```

Comment:
In the `updatedDataset`, some rules don't apply to the agent to which we are giving resource access, so this first step is just to filter out these and only check that the rules actually including the target agent don't include excessive access.

Reply:
OK, I think I understand what the issue is now:

1. We call `removeAgentFromResourceRule` to make sure that any existing rules for this agent-resource combination are removed, which splits up a rule into a rule for the agent (with different resources), and a rule for the resource (with different agents/groups/classes).
2. However, if the rule does not pertain to the given agent at all, creating a new rule for the agent (with different resources) means that all access given by that rule to other resources, will apply that access to the given agent.

Wouldn't the resolution here simply be to not split up the original rule if it does not apply to the given agent or to the given resource? Feels like it'd be easier to follow and safer that creating a new rule that possibly gives the agent too much access, and then remove that specific access again.

Reply:
You are right, it makes more sense to check if the agent is present in the rule before duplicating it rather than after to remove undue access. Fixed in 886ee47a9726eed22ec221d18e85a2ea4e8af64f.